com/williamfiset/algorithms/sorting/ShellSort.java

```diff
@@ -0,0 +1,60 @@
+// Java implementation of ShellSort 
+class ShellSort 
+{ 
+	/* An utility function to print array of size n*/
+	static void printArray(int arr[]) 
+	{ 
+		int n = arr.length; 
+		for (int i=0; i<n; ++i) 
+			System.out.print(arr[i] + " "); 
+		System.out.println(); 
+	} 
+
+	/* function to sort arr using shellSort */
+	int sort(int arr[]) 
```

Comment:
I would change the definition to public static int sort(int[] arr) so that one can call the method like ShellSort.sort(myarray);